src/components/molecules/EarthEditor/PropertyPane/PropertyField/SliderField/index.tsx

```diff
@@ -0,0 +1,216 @@
+import RCSlider from "rc-slider";
+import React, { useState, useEffect, useCallback, useRef } from "react";
+
+import Flex from "@reearth/components/atoms/Flex";
+import theme, { styled, metrics } from "@reearth/theme";
+import { metricsSizes } from "@reearth/theme/metrics";
+
+import { FieldProps } from "../types";
+
+export type Props = FieldProps<number> & {
+  min?: number;
+  max?: number;
+  step?: number;
+};
+
+const inputRegex = new RegExp(/^\d+\.\d{2,}$/);
+
+const SliderField: React.FC<Props> = ({
+  value,
+  linked,
+  overridden,
+  disabled,
+  onChange,
+  min,
+  max,
+  step,
+}) => {
+  const [inputValue, setInputValue] = useState(value);
+  const [sliderValue, setSliderValue] = useState(value);
+  const isEditing = useRef(false);
+  const isDirty = useRef(false);
+
+  const calculatedStep = step ? step : max ? max / 10 : 0.1;
+
+  const opacityMarkers = {
+    [min ?? 0]: min ?? 0,
+    [max ? max / 2 : 0.5]: max ? max / 2 : 0.5,
+    [max ?? 1]: max ?? 1,
+  };
```

Comment:
```suggestion
  const opacityMarkers = useMemo(() => ({
    [min ?? 0]: min ?? 0,
    [max ? max / 2 : 0.5]: max ? max / 2 : 0.5,
    [max ?? 1]: max ?? 1,
  }), [min, max]);
```

useMemo is just helpful here